src/SWP/Component/Revision/Repository/RevisionRepositoryInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Superdesk Web Publisher Templates System.
+ *
+ * Copyright 2015 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Component\Revision\Repository;
+
+use Doctrine\ORM\QueryBuilder;
+
+interface RevisionRepositoryInterface
+{
+    /**
+     * @return QueryBuilder
+     */
+    public function getPublishedRevision(): QueryBuilder;
```

Comment:
not sure if it should return QueryBuilder, it should return revision object or null, this way the repository interface can stay in the component and it will not have a dependency on QueryBuilder. The same for other methods in this class.